libraries/AP_Proximity/AP_Proximity_Backend.cpp

```diff
@@ -99,23 +101,28 @@ bool AP_Proximity_Backend::database_prepare_for_push(Vector3f &current_pos, Matr
     body_to_ned = AP::ahrs().get_rotation_body_to_ned();
 
     return true;
+#else
+    return false;
+#endif
 }
 
 // update Object Avoidance database with Earth-frame point
-void AP_Proximity_Backend::database_push(float angle, float distance)
+void AP_Proximity_Backend::database_push(float angle, float pitch, float distance)
 {
     Vector3f current_pos;
     Matrix3f body_to_ned;
 
     if (database_prepare_for_push(current_pos, body_to_ned)) {
-        database_push(angle, distance, AP_HAL::millis(), current_pos, body_to_ned);
+        database_push(angle, pitch, distance, AP_HAL::millis(), current_pos, body_to_ned);
     }
 }
 
 // update Object Avoidance database with Earth-frame point
 // pitch can be optionally provided if needed
 void AP_Proximity_Backend::database_push(float angle, float pitch, float distance, uint32_t timestamp_ms, const Vector3f &current_pos, const Matrix3f &body_to_ned)
 {
+
+#if !APM_BUILD_TYPE(APM_BUILD_AP_Periph)
```

Comment:
```suggestion
#if AP_OADATABASE_ENABLED
```

and all related changes

Reply:
If it's okay I would like to do this as another PR to not increase the scope of this one